### ISSUE TITLE
hubot-deploy requests too much access

### ISSUE THREAD

**Issue by benmanns** [atmos/hubot-deploy#77]
The hubot-deploy configuration [instructs the user](https://github.com/atmos/hubot-deploy/blob/master/docs/configuration.md#user-tokens) to provide `user`, `repo` access to their account. I don't see any way to limit the scope of access to a single organization, and I'd rather not grant access to all my private repositories to any organization using hubot-deploy.

I switched my token out for [`repo_deployment` scope](https://developer.github.com/v3/oauth/#scopes) which seems to still work properly. Is there a reason we need to ask for full access to `user` and `repo`? If not, can I update the documentation to specify the reduced scope?

**Comment by atmos** [comment]
You're right. I'm using the [archive link](https://developer.github.com/v3/repos/contents/#get-archive-link) for heroku and code deploy deployments too. Maybe we can change the requirement to be an env var?
